aidb/inference/bound_inference_service.py

```diff
@@ -193,19 +194,21 @@ async def _get_inputs_not_in_cache_table(self, inputs: pd.DataFrame, conn):
     if len(normalized_cache_cols) == 1:
         # For a single column, use `isin` and negate the condition
         col = normalized_cache_cols[0]
-        out_cache_df = inputs[~inputs[col].isin(cache_entries.index)]
-        out_cache_df_primary = out_cache_df[[col]]
+        is_in_cache = inputs[col].isin(cache_entries.index)
     else:
         # For multiple columns, create tuples and use set operations
         inputs_tuples = inputs[normalized_cache_cols].apply(tuple, axis=1)
         cache_tuples = set(cache_entries.index)
-        out_cache_df = inputs[~inputs_tuples.isin(cache_tuples)]
-        out_cache_df_primary = out_cache_df[normalized_cache_cols]
+        is_in_cache = inputs_tuples.isin(cache_tuples)
+
+    in_cache_df_primary = inputs[is_in_cache][normalized_cache_cols]
+    out_cache_df = inputs[~is_in_cache]
+    out_cache_df_primary = out_cache_df[normalized_cache_cols]
 
-    return out_cache_df, out_cache_df_primary
+    return out_cache_df, out_cache_df_primary, in_cache_df_primary
 
 
-  async def infer(self, inputs: pd.DataFrame):
+  async def infer(self, inputs: pd.DataFrame, if_return=False):
```

Comment:
this name is awkward, what does `if_return` mean?

Reply:
Renamed it to 'return_inference_results', which means if the 'infer' function should return the inference results.